api/storage_service.cc

```diff
@@ -681,9 +681,10 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         });
     });
 
-    ss::force_keyspace_compaction.set(r, [&ctx](std::unique_ptr<request> req) {
-        auto keyspace = validate_keyspace(ctx, req->param);
-        auto column_families = split_cf(req->get_query_param("cf"));
+    register_path_routine(ctx, r, "keyspace_compaction", ss::force_keyspace_compaction, [&ctx](std::unordered_map<sstring, sstring>&& val) {
+        sstring keyspace = val["ks"];;
```

Comment:
(here and below)
Validation is not needed? (sorry, just asking - better safe than sorry)
Also, extra semicolon.

Reply:
`register_path_routine` performs keyspace validation before calling the provided function and puts the validated string under `ks` in the new param map. That being said, I'm not sure it's that great of a solution, especially renaming the param - I don't see a good reason for doing so and it seems like something that can cause bugs in the future (people assume there's a keyspace under the `keyspace` key). Would you agree it's good to change it?

Reply:
> Would you agree it's good to change it?

I think so, but let's say it's of low priority ATM